src/app/pages/sharing/components/shares-dashboard/shares-dashboard.component.html

```diff
@@ -4,24 +4,68 @@
     ></ng-container>
+    <ng-container *ngIf="webdavTableConf?.tableExtraActions" headerCustomActions>
+      <button mat-icon-button [matMenuTriggerFor]="rowActions" aria-label="Table Actions of Expandable Table">
+        <mat-icon>more_vert</mat-icon>
+      </button>
+
+      <mat-menu #rowActions="matMenu">
+        <button mat-menu-item *ngFor="let action of webdavTableConf.tableExtraActions" (click)="action.onClick()">
+          {{action.label}}
+        </button>
+      </mat-menu>
+    </ng-container>
   </app-expandable-table>
   <app-expandable-table [ngClass]="getNfsOrder()" [expandableTableState]="nfsExpandableState" [disabled]="true" [conf]="nfsTableConf">
     <ng-container
       headerCustomContent
       *ngTemplateOutlet="nfsServiceStatus === ServiceStatus.Loading ? statusLoading : statusButton; context: { status: nfsServiceStatus, count: nfsHasItems }"
     ></ng-container>
+    <ng-container *ngIf="nfsTableConf?.tableExtraActions" headerCustomActions>
+      <button mat-icon-button [matMenuTriggerFor]="rowActions" aria-label="Table Actions of Expandable Table">
+        <mat-icon>more_vert</mat-icon>
+      </button>
+
+      <mat-menu #rowActions="matMenu">
+        <button mat-menu-item *ngFor="let action of nfsTableConf.tableExtraActions" (click)="action.onClick()">
+          {{action.label}}
+        </button>
+      </mat-menu>
+    </ng-container>
   </app-expandable-table>
   <app-expandable-table [ngClass]="getSmbOrder()" [expandableTableState]="smbExpandableState" [disabled]="true" [conf]="smbTableConf">
     <ng-container
       headerCustomContent
       *ngTemplateOutlet="smbServiceStatus === ServiceStatus.Loading ? statusLoading : statusButton; context: { status: smbServiceStatus, count: smbHasItems }"
     ></ng-container>
+    <ng-container *ngIf="smbTableConf?.tableExtraActions" headerCustomActions>
+      <button mat-icon-button [matMenuTriggerFor]="rowActions" aria-label="Table Actions of Expandable Table">
+        <mat-icon>more_vert</mat-icon>
+      </button>
+
+      <mat-menu #rowActions="matMenu">
+        <button mat-menu-item *ngFor="let action of smbTableConf.tableExtraActions" (click)="action.onClick()">
+          {{action.label}}
+        </button>
+      </mat-menu>
+    </ng-container>
   </app-expandable-table>
   <app-expandable-table [ngClass]="getIscsiOrder()" [expandableTableState]="iscsiExpandableState" [disabled]="true" [conf]="iscsiTableConf">
     <ng-container
       headerCustomContent
       *ngTemplateOutlet="iscsiServiceStatus === ServiceStatus.Loading ? statusLoading : statusButton; context: { status: iscsiServiceStatus, count: iscsiHasItems }"
     ></ng-container>
+    <ng-container *ngIf="iscsiTableConf?.tableExtraActions" headerCustomActions>
+      <button mat-icon-button [matMenuTriggerFor]="rowActions" aria-label="Table Actions of Expandable Table">
```

Comment:
Seems like `headerCustomActions` has a similar template across tables. Can we extract it to a common template and use ngTemplateOutlet with context to pass values?

Reply:
Fixed.